src/highdicom/seg/sop.py

```diff
@@ -944,3 +966,1458 @@ def _encode_pixels(self, planes: np.ndarray) -> bytes:
+
+    def search_for_segments(
+        self,
+        segment_label: Optional[str] = None,
+        segmented_property_category: Optional[Union[Code, CodedConcept]] = None,
+        segmented_property_type: Optional[Union[Code, CodedConcept]] = None,
+        algorithm_type: Optional[Union[SegmentAlgorithmTypeValues, str]] = None,
+        tracking_uid: Optional[str] = None,
+        tracking_id: Optional[str] = None,
+    ) -> List[int]:
+        """Get a list of segment numbers matching provided criteria.
+
+        Any number of optional filters may be provided. A segment must match
+        all provided filters to be included in the returned list.
+
+        Parameters
+        ----------
+        segment_label: Optional[str], optional
+            Segment label filter to apply.
+        segmented_property_category: Optional[Union[Code, CodedConcept]], optional
+            Segmented property category filter to apply.
+        segmented_property_type: Optional[Union[Code, CodedConcept]], optional
+            Segmented property type filter to apply.
+        algorithm_type: Optional[Union[SegmentAlgorithmTypeValues, str]], optional
+            Segmented property type filter to apply.
+        tracking_uid: Optional[str], optional
+            Tracking unique identifier filter to apply.
+        tracking_id: Optional[str], optional
+            Tracking identifier filter to apply.
+
+        Returns
+        -------
+        List[int]
+            List of all segment numbers matching the provided criteria.
+
+        """  # noqa: E501
+        filter_funcs = []
+        if segment_label is not None:
+            filter_funcs.append(
+                lambda desc: desc.segment_label == segment_label
+            )
+        if segmented_property_category is not None:
+            filter_funcs.append(
+                lambda desc:
+                desc.segmented_property_category == segmented_property_category
+            )
+        if segmented_property_type is not None:
+            filter_funcs.append(
+                lambda desc:
+                desc.segmented_property_type == segmented_property_type
+            )
+        if algorithm_type is not None:
+            algo_type = SegmentAlgorithmTypeValues(algorithm_type)
+            filter_funcs.append(
+                lambda desc:
+                SegmentAlgorithmTypeValues(desc.algorithm_type) == algo_type
+            )
+        if tracking_uid is not None:
+            filter_funcs.append(
+                lambda desc: desc.tracking_uid == tracking_uid
+            )
+        if tracking_id is not None:
+            filter_funcs.append(
+                lambda desc: desc.tracking_id == tracking_id
+            )
+
+        matches = [
+            desc.segment_number
+            for desc in self.SegmentSequence
+            if all(f(desc) for f in filter_funcs)
+        ]
+
+        return matches
+
+    def search_for_tracking_ids(
```

Comment:
```suggestion
    def get_tracking_ids(
```

Reply:
I am wondering whether it would make sense to combine `get_tracking_ids()` and `get_tracking_uids()` into one function and return `Tuple[str, str]`. What do you think?

Reply:
Perhaps. The issue with this is what happens if the mapping of tracking ID to tracking UID is not 1-to-1? Would it return all unique pairs? But then some IDs or UIDs would be repeated. Additionally I think most users will be referencing segments by either tracking ID or tracking UID, and not both so will normally throw away half the returned information

Reply:
Changed to `get_tracking_ids` in df7cb4801c8e090870e30e90d29e992e0e62efb3

Reply:
After reading up on the standard and realising that there should be a one-to-one mapping between tracking id and tracking uid, I changed my mind and decided to merge into a single function as suggested